bluemira/power_cycle/time.py

```diff
@@ -204,250 +232,95 @@ class ScenarioBuilder:
-            element_specs = breakdown_config[element_label]
-
-            element_name = element_specs["name"]
-            module = element_specs["module"]
-            variables_map = element_specs["variables_map"]
-
-            duration = cls.import_duration(module, variables_map)
-            breakdown_library[element_label] = {element_name: duration}
-        return breakdown_library
-
-    @staticmethod
-    def _build_phase_breakdown(breakdown_library, breakdown_list, operator):
-        phase_breakdown = dict()
-        last_value = 0
-        for label in breakdown_list:
             try:
-                element = breakdown_library[label]
-            except KeyError:
-                raise ScenarioBuilderError(
-                    "library",
-                    f"Breakdown element {label!r} has not been defined.",
+                duration = raw_uc(
+                    variables_map["duration"], variables_map["unit"], "second"
                 )
-
-            element_key = list(element.keys())
-            element_key = element_key[0]
-            element_value = element[element_key]
-            if operator == "&":
-                pass
-            elif operator == "|":
-                if element_value > last_value:
-                    phase_breakdown = dict()
-                    last_value = element_value
-            else:
-                raise ScenarioBuilderError(
-                    "operator",
-                    f"Unknown routine for {operator!r} operator.",
+            except KeyError:
+                raise PowerCycleError(
+                    msg=f'Variables map incomplete no keys {set(("duration", "unit")) - variables_map.keys()}'
                 )
-            phase_breakdown[element_key] = element_value
-
-        return phase_breakdown
-
-    @classmethod
-    def _build_phase_library(cls, phase_config, breakdown_library):
-        phase_library = Library(PowerCyclePhase)
-        for phase_label in phase_config.keys():
-            phase_specs = phase_config[phase_label]
-            phase_name = phase_specs["name"]
+        return duration
 
-            breakdown_operator = phase_specs["logical"]
-            breakdown_list = phase_specs["breakdown"]
-            phase_breakdown = cls._build_phase_breakdown(
-                breakdown_library, breakdown_list, breakdown_operator
+    def _build_phase_library(self):
+        self._phase_library = {
+            k: PowerCyclePhase(
+                v.name,
+                self._build_phase_logical(
+                    [self._breakdown_library[br].duration for br in v.breakdown],
+                    v.logical,
+                ),
+                k,
             )
+            for k, v in self.scenario_config.phase_library.items()
+        }
 
-            phase = PowerCyclePhase(
-                phase_name,
-                phase_breakdown,
-                label=phase_label,
+    def _build_phase_logical(self, durations, operator) -> float:
```

Comment:
I agree that "&" and "|" are a bit opaque. I think some enum of defined options is probably the best route. It can have descriptive names and also gives you more options than binary as you said above could be limiting.